src/CheckoutSdk/Reconcilation/ReconciliationQueryPaymentsFilter.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Checkout.Common;
+using Checkout.Common.Extensions;
+using System;
+
+namespace Checkout.Reconciliation
+{
+    public sealed class ReconciliationQueryPaymentsFilter : IEquatable<ReconciliationQueryPaymentsFilter>
```

Comment:
What do you think to extend `QueryFilterDateRange.cs` and remove the `From` & `To` here?

Reply:
In this case from and to are optional. is From? and to?, queryFilterDateRange has the same parameter but are not optional.
Using inheritance only for a property inheritance on records (filters) destinated class is not recommended on C# because you are limitating the inheritance to that object only to complain on properties, there is no multiple inheritance. 
Is more common to define a IQueryFilterDateRange or somethign like IDateRangeQueriable, that command to implement that kind of filter.